.github/workflows/pre-commit-ci.yml

```diff
@@ -0,0 +1,36 @@
+
+# This workflow runs pre-commit hooks on pushes and pull requests to main
+# to enforce coding style. To ensure correct configuration, please refer to:
+#  https://lincc-ppt.readthedocs.io/en/latest/practices/ci_precommit.html
+name: Run pre-commit hooks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  pre-commit-ci:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+    - name: Install dependencies
```

Comment:
The dependency on mdet/ngmix/etc. makes this not work. For now we need to conda install things.